File: dotnet/ThirtyFour.UI/Table/ChildElementCellParser.cs
using OpenQA.Selenium;

namespace ThirtyFour.UI.Table
{
    /// <summary>
    /// Parsers a child element from a table cell
    /// </summary>
    public class ChildElementCellParser : ICellParser<IWebElement>
    {
        protected By locator;

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="childLocator">the locator for the child element</param>
        public ChildElementCellParser(By childLocator)
        {
            locator = childLocator;
        }

        /// <summary>
        /// Returns a child element from a table cell
        /// </summary>
        /// <param name="cell">The cell to parse</param>
        /// <returns>The child of the cell</returns>
        public IWebElement GetValue(IWebElement cell)
        {
            return cell.FindElement(locator);
        }
    }
}

File: java/src/main/java/com/github/danhumphrey/thirtyfour/ui/table/ChildElementCellParser.java
package com.github.danhumphrey.thirtyfour.ui.table;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

/**
 * Parsers a child element from a table cell
 *
 */
public class ChildElementCellParser implements CellParser<WebElement> {

	protected By childLocator;
	
	/**
	 * Constructor
	 * @param childLocator the locator used to find the child element
	 */
	public ChildElementCellParser(By childLocator) {
		this.childLocator = childLocator;
	}
		
	/**
	 * returns a child element from the cell
	 * @param cell the cell to parse
	 * @return WebElement the child element
	 */
	@Override
	public WebElement getValue(WebElement cell) {
		return cell.findElement(childLocator);
	}

}
